sklearn/manifold/_locally_linear.py

```diff
@@ -723,8 +724,7 @@ def transform(self, X):
         X = check_array(X)
         ind = self.nbrs_.kneighbors(X, n_neighbors=self.n_neighbors,
                                     return_distance=False)
-        weights = barycenter_weights(X, self.nbrs_._fit_X[ind],
-                                     reg=self.reg)
+        weights = barycenter_weights(X, ind, reg=self.reg)
```

Comment:
This is the cause of the test failure that only happens in the transform method: one used to pass `self.nbrs_._fit_X` and the new code passes `X` instead (the test data).

So indeed we need to pass `barycenter_weights(X, Y, ind, reg=1e-3)` and compute the distances between `X[i]` and `Y[ind]` instead.

- At fit time: `X == Y == X_train`
- At transform time: `X == X_test` and `Y == self.nbrs_._fit_X`